### ISSUE TITLE
Monkey Island 2 Special Edition wont start (32460)

### ISSUE THREAD

**Issue by 0d1n89** [ValveSoftware/Proton#2163]
# Compatibility Report
- Name of the game with compatibility issues: Monkey Island 2 Special Edition
- Steam AppID of the game: 32460

## System Information
- GPU: Intel I7 7700k
- Driver/LLVM version: Nvidia 415.25
- Kernel version: 4.19.12-arch1-1-ARCH
- Link to full system information report as [Gist](https://gist.github.com/):
- Proton version: 3.7 (but tried all)

Systeminformation: https://gist.github.com/0d1n89/45d74ed4edd1ec1c4636b48ba382143c

## I confirm:
- [x ] that I haven't found an existing compatibility report for this game.
- [ x] that I have checked whether there are updates for my system available.

Log:
[steam-32460.log](https://github.com/ValveSoftware/Proton/files/2717121/steam-32460.log)

## Symptoms

Hello,

i just tried steam with Proton with my Arch Linux, i7-7700k and an Nvidia GTX950 with the latest driver. I tried Monkey Island 2 Special Edition and everytime i start the game i got the message "Extraction failure. Unable to find a volume for file extraction. Please verify that you have proper permissions." First i had XFS as filesystem but i switched to EXT4 and have the same error. The Proton log is uploaded. Any ideas how to fix it?

0d1n89

**Comment by ghost** [comment]
> tried Monkey Island 2 Special Edition and everytime i start the game i got the message "Extraction failure. Unable to find a volume for file extraction. Please verify that you have proper permissions." First i had XFS as filesystem but i switched to EXT4 and have the same error. The Proton log is uploaded. Any ideas how to fix it?

Well, the game is one that does work for people (I just tested it as working fine). On top of that if you have the same error on two file systems then either the way you changed it wasn't the appropriate way or your installation of the game, Steam, or should something else be required here still isn't appropriate.

Your log shows various Exceptions and mine shows mostly the same (even though the game starts fine) with some differences including:

`Unhandled exception: page fault on read access to 0x00000000 in 32-bit code (0x004d89d2)`

and that's where's yours goes horribly wrong. So it looks like you still have some problems either with the file system, memory, files, or something.

In the end, you should not have the same error if it was a result of using XFS. This would mean that something is still messed up.

**Comment by 0d1n89** [comment]
Thank you for the answer.
I just installed Arch completely new today and still get the same error. It appears when proton tries to install the Microsoft VC Redist Package. I dont understand why i have no permissions

**Comment by ghost** [comment]
> Thank you for the answer.
> I just installed Arch completely new today and still get the same error. It appears when proton tries to install the Microsoft VC Redist Package. I dont understand why i have no permissions

It would be interesting if a clean install of Ubuntu or Fedora would work. If so then it would be packages or distro difference as the cause. If not then Proton/Wine has a problem with the hardware - which I don't see any reason why it would.

So based on that trying the opposite version of Steam might do something. Flatpack vs regular.

And if its not that then the way Arch is setup might not be fully compatible and needs something. If its throwing a permission error on you then that's where to start, but I also don't know why. Make sure the path and files have proper perms. Mine for example "lrwxrwxrwx. 1 1000 1000 32 .steam/steam -> /home/user/.local/share/Steam".

**Comment by 0d1n89** [comment]
I will install ubuntu if i find a partition but that should not be a problem.
Before that i try flatpak. 
The permissions are ok. Im the owner, i can create and delete files. I dont understand this message :(

**Comment by ghost** [comment]
> I will install ubuntu if i find a partition but that should not be a problem.
> Before that i try flatpak.
> The permissions are ok. Im the owner, i can create and delete files. I dont understand this message :(

I feel your pain =)

**Comment by 0d1n89** [comment]
So i tried flatpak and it worked. I dont know why but it does. 
The problem is that flatpak installs to my home folder and thats not big enough. I cant take another directory. i searched in the web and i found that i should create a file in /etc/flatpak/installations.d/ . But when i do this i am not able to install steam with flatpak... :(

edit: i just mount my Folder from the Hdd with "mount -o bind" and it worked

**Comment by dietyyli** [comment]
I have the same problem on Fedora 33. Steam is from rpmfusion and I did try flatpak too, but it still wont start.

Reason why it wont work for me might be different, but end result is the same it does not work :/

[steamlog.txt](https://github.com/ValveSoftware/Proton/files/6836104/steamlog.txt)
[steamflatpaklog.txt](https://github.com/ValveSoftware/Proton/files/6836105/steamflatpaklog.txt)

**Comment by dietyyli** [comment]
For what is worth https://forum.winehq.org/viewtopic.php?t=31763&p=120133

**Comment by dietyyli** [comment]
I guess nobody cares about this old game anymore? Especially since it works for most user. Monkey Island 1 works for me too, but not Monkey Island 2. 
On my logs there is that same Unhandled exception: page fault on read access to 0x00000000 in 32-bit code (0x004d89d2)

Any hints what I could try?
(I even tried using proton-ge but got same result :( )

**Comment by dietyyli** [comment]
One more comment. I tried the GOG version with WINE. At first it failed with same Unhandled exception but, after some winetricks it worked. I've been playing few hours now and everything seems to work. Just that someone else might find this useful, here are the winetricks I used. (not sure if all are necessary)

comctl32
d3dcompiler_42
d3dx9_42
dsound
faudio
msvcrt40
xact

**Comment by ipr** [comment]
With Proton 6.3-8 it just works (for me), no need for winetricks or any of that.
Can you re-test and see if you still have issues? (Clear old modifications if any.)

**Comment by dietyyli** [comment]
Still not working. But, this is not fresh install. I'm currently using Fedora 35 and I've upgraded several Fedora versions. So it is possible there is something broken in my install. Monkey Island 1 works. I'm using Nvidia driver 510.47.03

**Comment by ipr** [comment]
Can you run it with PROTON_LOG=1 with the recent version of Proton and post the log from that? Those other logs are kinda old and not exactly what would be useful.

**Comment by dietyyli** [comment]
[steam-32460.log](https://github.com/ValveSoftware/Proton/files/8098536/steam-32460.log)

New log. I see the same error.
Unhandled exception: page fault on read access to 0x00000000 in 32-bit code (0x004d89d2).

**Comment by iqqmuT** [comment]
[steam-32460.log](https://github.com/ValveSoftware/Proton/files/8111636/steam-32460.log)

Same crash here. Proton 6.3-8, no winetricks used.

**Comment by iqqmuT** [comment]
I installed `d3dcompiler_47` via protontricks and launched the game with `PROTON_USE_WINED3D=1 %command%` and then it worked.

**Comment by dietyyli** [comment]
> I installed `d3dcompiler_47` via protontricks and launched the game with `PROTON_USE_WINED3D=1 %command%` and then it worked.

Yes. `d3dcompiler_47` via protontricks and it works for me too now.

**Comment by Babeetlebum** [comment]
Sorry if this isn't the right place but I tested the game on steamdeck and it launches fine but the game is not playable because the menu keeps opening. Other deck users have reported having the same issue. The game works in desktop mode but with the new monkey island game coming up I thought more people would try to install the old games on their deck and it would be a good idea to figure out what's going on

I tried a few proton launch options (`PROTON_USE_WINED3D`, `PROTON_NO_ESYNC`, `PROTON_NO_FSYNC`) with no avail.

I found a very old thread saying this bug may be caused by multi-threading and forcing single thread would help but I don't know how to do that with proton

Here's a log:
[steam-32460.log](https://github.com/ValveSoftware/Proton/files/8425686/steam-32460.log)

If you have any idea for a workaround please hit me up

**Comment by Blisto91** [comment]
Which menu is it that comes up? Can you post a screenshot? 👀

**Comment by Babeetlebum** [comment]
> Which menu is it that comes up? Can you post a screenshot? eyes

It's the main menu, with "Resume", "Save", "Load" etc.

Here's a video, I'm pressing the "A" button here to select "Play new game" and then repeatedly selecting "Resume game"

https://user-images.githubusercontent.com/5504984/161971288-c7b02435-af06-4747-b1c3-32dc952544c0.mp4

**Comment by Blisto91** [comment]
Strange. Wonder if the game think you are pressing escape or something?
You said it doesn't happen in desktop mode? I'm am not completly sure about how the gaming ui works on the steam deck and unsure if the issue should be filed at [the steam client repo](https://github.com/ValveSoftware/steam-for-linux) or [at the gamescope repo.](https://github.com/Plagman/gamescope)
Maybe @kisak-valve has some input.

**Comment by rodrigo-kayala** [comment]
Just to report that the game fails to start with latest Proton Experimental. Works well with Proton 7.0-3.
[steam-32460.log](https://github.com/ValveSoftware/Proton/files/9027132/steam-32460.log)
[sysinfo.txt](https://github.com/ValveSoftware/Proton/files/9027133/sysinfo.txt)
